icalevents/icalparser.py

```diff
@@ -418,6 +437,11 @@ def parse_rrule(component, tz=UTC):
         if type(rdtstart) is datetime:
             rdtstart = normalize(rdtstart, tz=tz)
 
+        if type(rdtstart) is date:
+            for index, rru in enumerate(rrules):
+                if 'UNTIL' in rru:
+                    rrules[index]['UNTIL'] = [dt.date() if type(dt) is datetime else dt for dt in rru['UNTIL']]
```

Comment:
Could you add a comment explaining why we do this?

Reply:
I am using adjust_timezone now and I added a comment